### ISSUE TITLE
can't recognize primary key  

### ISSUE THREAD

**Issue by minyoung90** [sqlalchemy/sqlalchemy2-stubs#201]
**Describe the bug**
mypy can't recognize primary key defiend as function style

**Expected behavior**
I think mypy should catch all the information about defined columns

**To Reproduce**
I defined the function like (two functions are same thing)

```py
STR_PK: Callable[[str], Column] = lambda comment: Column(
    String(255), comment=comment, primary_key=True
)


#def STR_PK(comment: str, max_bytes: int = 255) -> Column[String]:
#    return Column(String(max_bytes), comment=comment, primary_key=True)
```

and my model is like...

```py
@as_declarative()
class Base:
    __name__: str
    __table_args__ = {"mysql_charset": "utf8mb4"}

    metadata = MetaData(schema=settings.DATABASE_DB_SCHEMA)

    def __init__(self, *args: Any, **kwargs: Any) -> None:
        ...


class A(Base):
    id = STR_PK("test pk")
```

Actually, The code is working well... but only problem is to give mypy errors.

**Error**

```
Unexpected keyword argument "id" for "A" mypy
```

**Versions.**
 - OS: ubuntu 20.04 lts
 - Python: 3.9.4
 - SQLAlchemy: 1.4.30
 - mypy: 0.930
 - SQLAlchemy2-stubs: 0.0.2a19

**Additional context**

**Have a nice day!**

**Comment by zzzeek** [comment]
this looks like the same thing as #8.   these issues will all be fixed by https://github.com/sqlalchemy/sqlalchemy/issues/7535 which no longer uses a plugin.

**Comment by minyoung90** [comment]
> this looks like the same thing as #8. these issues will all be fixed by [sqlalchemy/sqlalchemy#7535](https://github.com/sqlalchemy/sqlalchemy/issues/7535) which no longer uses a plugin.

so, do I have to upgrade sqlalchemy 1.4 to 2.0 ?
I can't find the information about 'mapped_column'. sqlalchemy 2.0 api document have it but, not much information about it.

How about this way? using '_mypy_mapped_attrs' works well... but, I think this way is not good...

```py
class Test(Base):
    # Relationships
    id: str = STR_PK("primary key")

    # Basics
    description: str = sa.Column(sa.Text, comment="description", nullable=False)

    # Mypy Helper
    if TYPE_CHECKING:
        _mypy_mapped_attrs = [id, description]
```

**Comment by zzzeek** [comment]
sqlalchemy 2.0 is in development right now so there isn't general availability of the new approach.

i think if you annotated your STR_PK as returning ``-> Mapped[str]`` that might be all you need.   that's more or less what the new thing will do also, just more completely.
